### DISCUSSION TITLE
`vercel build` failing using `node@20.x`

### DISCUSSION THREAD

**Discussion by Maxwell2022** [vercel/community#4973] [Help] [answered]
### Steps to Reproduce

1. update Vercel project general settings to use `node@20.x (beta)` (from 18.x)
2. update package.json `engine.node` value to `20.x`
3. run CI build (github action) which is doing `vercel pull` and `vercel build`
4. notice `vercel build` is failing with the following error:

> Vercel CLI 32.5.4
Retrieving project…
> Downloading `preview` Environment Variables for Project X
Downloading
Created .vercel/.env.preview.local file  [99ms]

> Downloading project settings
Downloaded project settings to ~/work/X/X/.vercel/project.json [0ms]
Vercel CLI 32.5.4
Detected ENABLE_EXPERIMENTAL_COREPACK=1 and "pnpm@8.11.0" in package.json
Error: Found invalid Node.js Version: "20.x". Please set Node.js Version to 18.x in your Project Settings to use Node.js 18.
Learn More: http://vercel.link/node-version
Error: Process completed with exit code 1.

**Comment by swarnava** [comment, marked as answer]
Hi, Can you upgrade your vercel CLI and give it a try?

**Reply by Maxwell2022** [reply]
yes, that was the issue. Thanks

**Reply by Maxwell2022** [reply]
I left a feedback on `http://vercel.link/node-version` to improve the documentation page to perform this check.
It probably makes sense since it's where we redirect users in case this is happening
